Shared/Supporting Files/Models/OnDemandResource.swift

```diff
@@ -35,54 +35,59 @@ final class OnDemandResource: ObservableObject {
     /// The progress of the on-demand resource request.
     var progress: Progress { request.progress }
     
+    /// A Boolean value indicating whether a resource request can be initiated.
+    var isDownloadable: Bool {
+        requestState != .inProgress && requestState != .downloaded
+    }
+    
     /// The current state of the on-demand resource request.
-    @Published private(set) var requestState: RequestState = .notStarted
+    private(set) var requestState: RequestState?
     
     /// The error occurred in downloading resources.
-    @Published private(set) var error: (any Error)?
+    private(set) var error: (any Error)?
     
     /// The on-demand resource request.
     private let request: NSBundleResourceRequest
     
-    /// A set of cancellable instances for the request progress subscription.
-    private var cancellables: Set<AnyCancellable> = []
+    /// The name of the sample associated with the resource request.
+    let sampleName: String
     
-    /// Initializes a request with a set of Resource Tags.
-    init(tags: Set<String>) {
-        request = NSBundleResourceRequest(tags: tags)
+    /// Initializes a request with a sample.
+    init(sample: Sample) {
```

Comment:
This type was intentionally designed to be sample agnostic. Figuring out the tags to use for a given sample shouldn't be part of its job. Instead, the tags should be passed to it like before and the name should be stored elsewhere, alongside the resource.